Add version locking to YUM commands

hakonhall opened the pull request.

hakonhall committed:
Add version locking to YUM commands

freva reviewed (approved):
Add a test where the command is on one of the other forms (`name-ver-rel`)?

hakonhall committed:
Require digit in version and release

hakonhall committed:
Extract YumPackageName

hakonhall commented:
Added a few more tests, and extracted YumPackageName into its own file to make the builder more readily available to clients, since the String syntax is so ambiguous.

PTAL

freva reviewed (commented):
Also please add alternative `install()/remove()/upgrade()` that takes in `YumPackageName...`

hakonhall commented on node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumPackageName.java:
```diff
@@ -0,0 +1,148 @@
+ *
+ * @author hakonhall
+ */
+public class YumPackageName {
```
Done

hakonhall commented:
> Also please add alternative install()/remove()/upgrade() that takes in YumPackageName

In this case, let me extend YPN to parse all existing packages so the internals can use YPN instead of String.

hakonhall committed:
Use YumPackageName internally

 - Make install, remove, etc also work with YumPackageName.
 - adds hashCode and equals

hakonhall commented:
PTAL @freva
